pkg/apis/kubeflow/v2beta1/types.go

```diff
@@ -131,6 +139,16 @@ type RunPolicy struct {
 	// Defaults to false.
 	// +kubebuilder:default:=false
 	Suspend *bool `json:"suspend,omitempty"`
+
+	// ManagedBy is used to indicate the controller or entity that manages a MPIJob.
+	// The value must be either an empty, 'kubeflow.org/mpi-operator' or
+	// 'kueue.x-k8s.io/multikueue'.
```

Comment:
This API cannot say that these are the only possible values. This API shouldn't even be aware that Kueue exists.

The only thing that it could say is that if the value is not `kubeflow.org/mpi-operator` then it's not managed by this controller. See https://github.com/kubernetes/api/blob/71385f038c1097af36f3d2f68b415860b866c1f8/batch/v1/types.go#L473 for an example of how it should look like.

Reply:
Here is background of the reason why we limit the acceptable values in the managedBy resource.
Ideally, we should enhance the status validations similar to batch/Job and allow managedBy field to be specified arbitrary values
so that we can provide the appropriate rail.

But, the proposal was rejected in the training-operator since the TrainigOperator community is working on the v2 API, and they have no enough bandwidth to implement the additional status validation implementation. And, the v2 API will provided the non restricted managedBy feature same as batch/Job.

So, I think we can take either approach: 1. enhance status validation and allow managedBy to specify arbitrary values 2. Limit the acceptable values kueue or mpi-operator.

If anyone can lead the design and handle the status validation enhancement, I'm ok with opt 1.

Reply:
I see, that's unfortunate.

Let's just do the same as the training-operator.

For future reference, the status validations refer to the equivalent of this https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/4368-support-managed-by-for-batch-jobs#job-status-validation

Reply:
Sorry, I forgot to add the contextual issue 😓 
https://github.com/kubeflow/training-operator/issues/2193

Reply:
> I see, that's unfortunate.
> 
> Let's just do the same as the training-operator.
> 
> For future reference, the status validations refer to the equivalent of this https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/4368-support-managed-by-for-batch-jobs#job-status-validation

Yes, ideally, we should do it. Maybe, we can leave the dream on the TrainJob 😓